extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultTransitSecretEngine.java

```diff
@@ -0,0 +1,90 @@
+import io.quarkus.vault.transit.SigningRequest;
+import io.quarkus.vault.transit.SigningResult;
+import io.quarkus.vault.transit.TransitContext;
+import io.quarkus.vault.transit.VerificationRequest;
+import io.quarkus.vault.transit.VerificationResult;
+import io.quarkus.vault.transit.Verifyable;
+
+public interface VaultTransitSecretEngine {
+
+    // encrypt
+
+    /**
+     * Encrypt a regular string with a vault key configured in the transit secret engine.
+     * equivalent to:
+     * {@code encrypt(keyName, plaintext.getBytes("UTF-8"), Optional.empty());}
+     * <p>
+     * this method is usually used in conjunction with {@link #decryptString(String, String)}
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param plaintext the string to encrypt
+     * @return cipher text
+     */
+    String encryptString(String keyName, String plaintext);
+
+    /**
+     * Encrypt a regular string with a vault key configured in the transit secret engine.
+     * if the key does not exist, and the policy specifies a create capability the key will be lazily created
+     * (i.e. upsert). the key can be further customized by specifying transit encryption-key configuration
+     * properties. as vault encrypts bytes, and not strings, the plaintext string is converted to bytes using
+     * the UTF-8 charset, before being passed to the hashicorp vault encryption service.
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param plaintext the string to encrypt
+     * @param transitContext an optional context for key derivation
+     * @return cipher text
+     */
+    Ciphertext encrypt(String keyName, Plaintext plaintext, TransitContext transitContext);
+
+    /**
+     * Encrypt a list of elements. This will return a list of results, each being either an ciphertext or an error.
+     * Each element can be specify an optional context, allowing key derivation.
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param requests the list of elements to encrypt
+     * @return a list of result objects containing either a ciphertext or an error
+     */
+    List<EncryptionResult> encrypt(String keyName, List<EncryptionRequest> requests);
+
+    // decrypt
+
+    String decryptString(String keyName, String ciphertext);
+
+    Plaintext decrypt(String keyName, Ciphertext ciphertext, TransitContext transitContext);
+
+    List<DecryptionResult> decrypt(String keyName, List<DecryptionRequest> requests);
+
+    // rewrap
+
+    Ciphertext rewrap(String keyName, Ciphertext ciphertext, TransitContext transitContext);
```

Comment:
@vsevel  If we do have `rewrap` from the very start, then lets have a simple variant without `transitContext`, same as with `encrypt` variations.

Reply:
I did not include it because it would read: `EncryptedData rewrap(String keyName, EncryptedData encryptedData);`
Whereas for `encrypt` It is not about the `transitContext`, but more about the fact that we take a `String` and turn it into `UTF-8` bytes, and return the cipher text as string.
So I think the condensed signature for `rewrap` seems less relevant.
What do you think?

Reply:
Lets revisit rewrap at the very end again